tests/unit/test_query_language.py

```diff
@@ -322,20 +323,36 @@ class some_table(EventFrame):
             some_int = Series(int)
 
 
-def test_table_from_file(tmp_path):
-    rows = [(1, 100), (2, 200)]
-    csv_path = tmp_path / "test.csv"
-    with csv_path.open("w") as f:
-        writer = csv.writer(f)
-        writer.writerow(("patient_id", "n"))
-        writer.writerows(rows)
+@pytest.mark.parametrize("file_extension", ["csv", "csv.gz", ".arrow"])
```

Comment:
Thinking about it further, you could actually do:
```py
from databuilder.file_formats import FILE_FORMATS

...

@pytest.mark.parametrize("file_extension", FILE_FORMATS.keys())
```
Which would mean that it's automatically parameterised over all formats we support (including any new ones we add).

The only thing to watch out for is that these extensions already include the leading dot so you'd need to change the name string to `f"test_file{file_extension}"`.

Reply:
ooh, I like this.